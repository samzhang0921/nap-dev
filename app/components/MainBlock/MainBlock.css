.MainBlock {
  box-sizing: border-box;
  border-bottom: 1px dotted #999;
  margin-bottom: 30px;
}
.clear {
  clear: both;
}
