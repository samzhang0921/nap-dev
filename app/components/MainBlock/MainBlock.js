import React from 'react';
import style from './MainBlock.css';
import SideFilter from '../SideFilter/SideFilter';
import ProductsList from '../ProductsList/ProductsList';

const MainBlock = props => {
    return (
      <div className={style.MainBlock}>
      <SideFilter designerList = {props.data.designers} updateCategory = {props.updateCategory}/>
      <ProductsList productViewChecked={props.productViewChecked} data = {props.data}/>

      <div className={style.clear}></div>
      </div>
    );
  }
export default MainBlock;
