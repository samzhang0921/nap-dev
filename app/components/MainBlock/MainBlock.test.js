import React from 'react';
import {mount} from 'enzyme';
import toJson from 'enzyme-to-json';
import renderer from 'react-test-renderer';
import MainBlock from './MainBlock';

describe('MainBlock', () => {

  it('should match its snapshot with items', () => {
    const data = {
      "offset": 0,
      "limit": 60,
      "total": 500,
      "designers": [
        {
          "id": 1643,
          "name": {
            "en": "A.P.C. Atelier de Production et de Création"
          }
        }
      ],
      "colors": [
        {
          "id": 2,
          "name": "black"
        }
      ],
      "products": [
        {
          "sku": 543002,
          "name": "Cashmere wrap",
          "price": "£380",
          "color": [57],
          "brand_name": "Madeleine Thompson",
          "brand_id": 1031,
          "caregories": [
            {
              "id": 5,
              "children": [
                {
                  "id": 4279,
                  "children": [
                    {
                      "id": 5166
                    }, {
                      "id": 5179
                    }
                  ]
                }
              ]
            }
          ],
          "image": {
            "outfit": "//cache.net-a-porter.com/images/products/543002/543002_ou_sl.jpg",
            "pview": "//cache.net-a-porter.com/images/products/543002/543002_in_sl.jpg"
          }
        }
      ]
    }
    const tree = renderer.create(<MainBlock data={data}/>).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
