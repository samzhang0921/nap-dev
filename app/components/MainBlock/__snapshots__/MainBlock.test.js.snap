// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainBlock should match its snapshot with items 1`] = `
<div
  className="MainBlock"
>
  <div
    className="sideFilter"
  >
    <div
      className="CatFilterList"
    >
      <div
        className="filterItem"
        data-category-id={0}
        onClick={[Function]}
      >
        Clothing
      </div>
      <div
        className="filterItem"
        data-category-id={40}
        onClick={[Function]}
      >
        Beachwear
      </div>
      <div
        className="filterItem"
        data-category-id={48}
        onClick={[Function]}
      >
        Coat
      </div>
      <div
        className="filterItem"
        data-category-id={46}
        onClick={[Function]}
      >
        Dress
      </div>
      <div
        className="filterItem"
        data-category-id={44}
        onClick={[Function]}
      >
        Jackets
      </div>
      <div
        className="filterItem"
        data-category-id={55}
        onClick={[Function]}
      >
        Jeans
      </div>
      <div
        className="filterItem"
        data-category-id={42}
        onClick={[Function]}
      >
        Pants
      </div>
      <div
        className="filterItem"
        data-category-id={39}
        onClick={[Function]}
      >
        Top
      </div>
      <div
        className="filterItem"
        data-category-id={49}
        onClick={[Function]}
      >
        Wrap
      </div>
    </div>
    <div
      className="DesignerFilter"
    >
      <h4>
        Designer
      </h4>
      <a
        className="clearFilter"
        onClick={[Function]}
      >
        Clear
      </a>
      <ul
        className="scrollPanel"
      >
        <li
          data-designer-id={1643}
          onClick={[Function]}
        >
          <span
            className="filterCheckbox"
          />
          <div
            className="designer"
          >
            <span>
              A.P.C. Atelier de Production et de Création
            </span>
          </div>
          <div />
        </li>
      </ul>
    </div>
    <div
      className="ColorFilter"
    >
      <h4>
         COLOR 
      </h4>
      <a
        className="clearFilter"
        onClick={[Function]}
      >
        Clear
      </a>
      <ul>
        <li
          data-category-id={2}
          onClick={[Function]}
        >
          <span
            className="colorCheckbox"
          />
          <span
            className="colorName"
          >
            black
          </span>
          <div
            className="clear"
          />
        </li>
      </ul>
    </div>
  </div>
  <div
    className="productsList"
  >
    <ul>
      <li
        className="listItem"
      >
        <div
          className="productImage"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <img
            src="//cache.net-a-porter.com/images/products/543002/543002_ou_sl.jpg"
          />
          <div
            className="quickView"
            onClick={[Function]}
          >
            Quick View
          </div>
        </div>
        <div
          className="productInfo"
        >
          <span
            className="designer"
            data-product-id={543002}
          >
            Madeleine Thompson
          </span>
          <br />
          Cashmere wrap
          <br />
          <span>
            £380
          </span>
          <br />
        </div>
      </li>
    </ul>
  </div>
</div>
`;
